src/query/functions/utils/group_test.go

```diff
@@ -230,6 +230,32 @@ var collectTest = []struct {
 			{"c": "3", "d": "6"},
 		}),
 	},
+	{
+		"metrics name",
+		[]string{"a"},
+		multiTagsFromMaps([]map[string]string{
+			{"__name__": "foo", "a": "1"},
+			{"__name__": "foo", "a": "1", "b": "2", "c": "4"},
+			{"__name__": "foo", "b": "2"},
+			{"__name__": "foo", "a": "1", "b": "2", "c": "3"},
+			{"__name__": "foo", "a": "1", "b": "2", "d": "3"},
+			{"__name__": "foo", "c": "d"},
+		}),
+		[][]int{{2, 5}, {0, 1, 3, 4}},
+		multiTagsFromMaps([]map[string]string{
+			{},
```

Comment:
nit: do we need the empty entry here?

Reply:
Yeah, it's an empty tags map resulting from tags removed from the first series by the `without`.